src/actions/index.ts

```diff
@@ -1,25 +1,48 @@
-import {Action} from 'redux';
+import {
+  TValuePayload,
+  TUpdateInArray,
+  TIndexPayload,
+  TSaveForm
+} from '../store/form/types';
+import {TPayloadAction} from '../store/types';
+import {TFetchRacesAligmentAction} from '../store/wizard/types';
 
-export interface IPayloadAction extends Action {
-  payload?: any;
-}
-
-export const saveForm = (form) => ({
+export const saveForm = ({path, value}: TSaveForm): TPayloadAction => ({
   type: 'SAVE_FORM',
-  payload: form
+  payload: {
+    path,
+    value
+  }
 });
 
-export const addSkill = () => ({ type: 'ADD_SKILL' });
+export const addIntoArray = ({path , value}: TValuePayload):
+  TPayloadAction => ({
+  type: 'PUSH_INTO_ARRAY',
```

Comment:
I think all of the references to an array in the names of our actions and reducers should probably be made more general. Don't need to change it right now but maybe we can find something that's more declarative of the intent, like `SAVE_FORM_WITH_INDEX` or something